test/buildifier.test.ts

```diff
@@ -29,32 +29,54 @@ const workspacePath = path.join(
 
 describe("buildifier", () => {
   it("diagnostics are added from buildifier", async () => {
+    // Promise that resolves when diagnostics are added to the file.
+    const diagnosticsPromise = new Promise<vscode.Diagnostic[]>(
+      (resolve, reject) => {
+        const timeout = setTimeout(() => {
+          disposable.dispose();
+          reject(new Error("Timed out waiting for diagnostics"));
+        }, 5000);
+
+        const disposable = vscode.languages.onDidChangeDiagnostics((e) => {
+          if (!e.uris.some((u) => u.toString() === uri.toString())) {
+            return;
+          }
+          const diags = vscode.languages.getDiagnostics(uri);
+          if (diags.length > 0) {
+            clearTimeout(timeout);
+            disposable.dispose();
+            resolve(diags);
+          }
+        });
+      },
+    );
+
+    // Create DiagnosticsManager and open file
     const buildFile = path.join(workspacePath, "buildifier", "BUILD");
+    const uri = vscode.Uri.file(buildFile);
+    disposables.push(new BuildifierDiagnosticsManager());
     await openSourceFile(buildFile);
 
-    disposables.push(new BuildifierDiagnosticsManager());
+    // Wait for diagnostics for this file
+    const diagnostics = await diagnosticsPromise;
 
-    const diagnostics = vscode.languages.getDiagnostics(
-      vscode.Uri.file(buildFile),
-    );
+    // Assert expected diagnostic is among returned diagnostics
+    const hasExpected = diagnostics.some((d) => {
```

Comment:
How come you've refactored the assertion like this? This will give much worse error messages when the assertion fails.